src/test/regress/expected/multi_extension.out

```diff
@@ -1332,6 +1332,20 @@ SELECT * FROM multi_extension.print_extension_changes();
 
 -- Test downgrade to 11.3-1 from 12.0-1
 ALTER EXTENSION citus UPDATE TO '12.0-1';
+CREATE TABLE null_shard_key (x int, y int);
+SELECT create_distributed_table('null_shard_key', null);
+ create_distributed_table
+---------------------------------------------------------------------
+
+(1 row)
+
+-- Show that we cannot downgrade to 11.3-1 becuase the cluster has a
+-- distributed table with single-shard.
+ALTER EXTENSION citus UPDATE TO '11.3-1';
+ERROR:  cannot downgrade Citus because there are distributed tables without a shard key.
+HINT:  To downgrade Citus to an older version, you should first convert those tables to Postgres tables by executing SELECT undistribute_table("%s").
```

Comment:
does that look like a good hint? @agedemenli @marcocitus

Reply:
I think it should include the name of the table. Otherwise people probably have no clue what the error means.

Reply:
Isn't it clear that user need to undistribute the distributed tables that don't have a shard key?

Reply:
I think most/some users won't realise that schema based sharding creates this distributed tables without a shard key. Probably we'll want to have some check for distributed schemas too.

With the current error people have no clear next steps if they don't know how to find distributed tables without a shard key.  Having a hint as to which table/schema we're erroring because is quite useful then. Another approach would be to give the query with which to find these tables/schemas.

Reply:
> I think most/some users won't realise that schema based sharding creates this distributed tables without a shard key. Probably we'll want to have some check for distributed schemas too.

Yeah, we're doing so in https://github.com/citusdata/citus/pull/6866.